### DISCUSSION TITLE
Using Autoload

### DISCUSSION THREAD

**Discussion by TonioIncredible** [PHPOffice/PhpSpreadsheet#3036] [Q&A] [answered]
Hi,

Maybe I am dumb, but I don't manage to fix this Autoload story.

I downloaded PhpSpreadsheet and inserted it in my project, but I don't know how to install correctly an autoload. Previously I was using PhpExcel and had no issue with it, but it does not work anymore (maybe due to PHP 7?).

I read the doc about composer and understood nothing...

I am not a Linux man.

My platform:

- Windows 10 64 bits
- Wampserver 3.1.4 with PHP 7.2.10
- Intellij Idea

If someone can help me, it will be wonderful.

Thanks in advance

**Comment by oleibman** [comment, marked as answer]
For PHPExcel, you had a statement near the start of your script like (your directory may vary):
```php
require __DIR__ . '/Classes/PHPExcel.php';
```
For PhpSpreadsheet, you need to use the following instead:
```php
require __DIR__ . '/vendor/autoload.php';
```

**Reply by TonioIncredible** [reply]
Thank you,

But where do I find this '/vendor/autoload.php' ? It is not coming with PhpSpreadsheet.

**Reply by MarkBaker** [reply]
PhpSpreadsheet is a `composer` install, as described in the [README instructions](https://github.com/PHPOffice/PhpSpreadsheet#installation). If you choose to download it manually rather than using composer, then you need to download all dependencies and set up autoloading manually

[These instructions](https://github.com/PHPOffice/PhpSpreadsheet/issues/31#issuecomment-354502740) may provide some help if you really want to do all this manually, but why don't you use composer?

[Composer](https://getcomposer.org/) is not a "linux thing", it is a "PHP thing", a command-line dependency management tool. When you do the install using composer, it installs all dependencies in a folder called `vendor`, and creates an autoloader file that will autoload all those dependencies

**Reply by TonioIncredible** [reply]
Thanks, I managed with composer.
